### PR TITLE
WIP: Added option to select bucket name for deployable assets

### PR DESCRIPTION
**Pull request by acere**
*Issue #, if available:*
#3 
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -13,10 +13,25 @@
 # Create App and stacks
 app = core.App()
 
+artifact_bucket = app.node.try_get_context("ArtifactBucket")
```
Comment (brightsparc) [inline app.py:16]: Can you add the "drift:" prefix as per the other `cdk.json` values.

File: app.py [hunk 2]
```diff
@@ -13,10 +13,25 @@
...
+if artifact_bucket is not None:
+    synth = core.DefaultStackSynthesizer(
+        file_assets_bucket_name=artifact_bucket,
+        generate_bootstrap_version_rule=False,
+    )
+PipelineStack(app, "drift-pipeline", synthesizer=synth)
 
 # Create the SC stack
-ServiceCatalogStack(app, "drift-service-catalog")
+synth = core.DefaultStackSynthesizer()
```
Comment (brightsparc) [inline app.py:28]: Do we need to declare this twice, or can we use the synth object?

File: app.py [hunk 3]
```diff
@@ -13,10 +13,25 @@
...
 app = core.App()
 
+artifact_bucket = app.node.try_get_context("ArtifactBucket")
+
 # Create the pipeline stack
-PipelineStack(app, "drift-pipeline")
+synth = core.DefaultStackSynthesizer()
+if artifact_bucket is not None:
+    synth = core.DefaultStackSynthesizer(
+        file_assets_bucket_name=artifact_bucket,
```
Comment (brightsparc) [inline app.py:22]: Could we also pass in `bucket_prefix` so that I can provide `drit-pipeline` in this case?

**Review by brightsparc — COMMENTED** [review]
Hey, i’ve been taking a closer look at this - perhaps there is an easier way which doesn’t require code change to publish to s3.  Looks like bootstrap has the ability to set `--toolkit-bucket-name` as well as `--public-access-block-configuration=false`  which should allow updating these files and making them public (which would be required for CFN templates).  It would be nice if we could specify the bucket prefix through config but that doesn’t look like an option: https://github.com/aws/aws-cdk/blob/master/design/cdk-bootstrap.md

File: app.py (hunk 1 above)
Comment (acere) [reply app.py]: Done

File: app.py (hunk 2 above)
Comment (acere) [reply app.py]: We need to declare it for every synth operation, we can't reuse it after "spending" it

File: app.py (hunk 3 above)
Comment (acere) [reply app.py]: Done!

**Pull request merged**